src/db.rs

```diff
@@ -71,6 +71,61 @@ impl DB {
         Ok(())
     }
 
+    pub fn migrate<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        if !self.dirs.is_empty() {
+            return Err(anyhow!(
+                "To prevent conflicts, you can only migrate from z with an empty \
+                 zoxide database!"
+            ));
+        }
+
+        let zdata = File::open(path)?;
+        let reader = BufReader::new(zdata);
+
+        for line in reader.lines() {
+            let line = line?;
+
+            let first_pipe = line
+                .find('|')
+                .with_context(|| anyhow!("missing separator"))?;
+            let last_pipe = line
+                .rfind('|')
+                .with_context(|| anyhow!("missing separator"))?;
+
+            if first_pipe == last_pipe {
+                return Err(anyhow!("invalid data file format -- only 1 separator"));
+            }
+
+            let path = PathBuf::from(&line[..first_pipe]);
+            let rank = line[first_pipe + 1..last_pipe]
+                .parse::<f64>()
+                .with_context(|| anyhow!("could not parse rank"))?;
+
+            // otherwise, the rank will get scaled down, depending on how old
+            // the entry is
+            let epoch = util::get_current_time()?;
+
+            let path_abs = match path.canonicalize() {
+                Ok(path) => path,
+                Err(_) => continue, // ignore dead paths
+            };
+
+            let path_str = path_abs
```

Comment:
You should not be using `String` to represent paths, use `OsString`

Reply:
(Covered this on IRC, but leaving it here for posterity/conversation's sake)

I agree. But I believe it is out of scope for this PR and should be done somewhere else (either a follow-up PR, a commit by our friendly neighborhood maintainer, or a PR prior to landing this).

Reply:
I think `PathBuf` would be more fitting, and `serde` supports it too. I kept it as a `String` for the initial release for simplicity of string matching, but I will change this in a later commit.